library/psa_crypto_slot_management.c

```diff
@@ -124,15 +124,35 @@ static psa_status_t psa_load_persistent_key_into_slot( psa_key_slot_t *p_slot )
     psa_status_t status = PSA_SUCCESS;
     uint8_t *key_data = NULL;
     size_t key_data_length = 0;
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
 
-    status = psa_load_persistent_key( p_slot->persistent_storage_id,
-                                      &( p_slot )->type,
-                                      &( p_slot )->policy, &key_data,
-                                      &key_data_length );
+    psa_set_key_id( &attributes, p_slot->persistent_storage_id );
+    status = psa_load_persistent_key( &attributes,
+                                      &key_data, &key_data_length );
     if( status != PSA_SUCCESS )
         goto exit;
-    status = psa_import_key_into_slot( p_slot,
-                                       key_data, key_data_length );
+    p_slot->lifetime = psa_get_key_lifetime( &attributes );
+    p_slot->type = psa_get_key_type( &attributes );
+    p_slot->policy = attributes.policy;
```

Comment:
Could we just store attributes in the slot? Then we could directly `psa_load_persistent_key()` into the `p_slot->attributes`

Reply:
I thought about this, as it would simplify the design. The thing is, the layout of `psa_key_attributes_t` may change without warning and that's just an ABI (not API) change, whereas the layout of storage must not change without providing an upgrade mechanism, otherwise you can't upgrade the code on an existing device. There may be a better way to do it (and it's even more visible in https://github.com/ARMmbed/mbed-crypto/pull/183 which also saves the bit-size and the slot number for SE keys), but that would be going too far for this PR.